runtime/expr.h

```diff
@@ -90,6 +90,10 @@ enum class intrinsic {
   // Common arithmetic functions.
   abs,
 
+  // These are short circuiting logical operations, unlike the && and || operators which are commutative and do not short circuit.
```

Comment:
The fact that && and || are commutative (unlike C++) is a bit of a footgun I guess but not enough to worry about

Reply:
It's too useful of a property for a theorem prover, which is really what we need from this. Halide is the same way (in addition to short circuiting being ill defined for SIMD).